colossalai/nn/optimizer/cpu_adamw.py

```diff
@@ -0,0 +1,124 @@
+import torch
+import math
+
+
+class CPUAdamW(torch.optim.Optimizer):
+    optimizer_id = 0
+
+    def __init__(self,
+                 model_params,
+                 lr=1e-3,
+                 bias_correction=True,
+                 betas=(0.9, 0.999),
+                 eps=1e-8,
+                 weight_decay=0,
+                 loss_scale=-1,
+                 simd_log=False):
+        """
+        An implementation equivalent to `torch.optim.Adam`.
+        The difference is that model_params are sharded parameters belonging to a ShardedModelV2 instance.
+        The sharded param of model_params can resident on both CPU and CUDA.
+        """
+
+        default_args = dict(lr=lr, betas=betas, eps=eps, weight_decay=weight_decay, bias_correction=bias_correction)
+        super(CPUAdamW, self).__init__(model_params, default_args)
+        self.opt_id = CPUAdamW.optimizer_id
+        CPUAdamW.optimizer_id = CPUAdamW.optimizer_id + 1
+        adamw_mode = True
+        self.loss_scale = loss_scale
+        try:
+            import cpu_adam
+        except ImportError:
+            raise ImportError('Please install colossalai from source code to use CPUAdam')
+        self.cpu_adam_op = cpu_adam
+        self.cpu_adam_op.create_adam(self.opt_id, lr, betas[0], betas[1], eps, weight_decay, adamw_mode, simd_log)
+
+    def __del__(self):
+        if self.cpu_adam_op:
+            self.cpu_adam_op.destroy_adam(self.opt_id)
+
+    def torch_adamw_update(self,
```

Comment:
Is it possible for us to use fused `Adam` here? @feifeibear @Gy-Lu

Reply:
ok, I'll try :)